lib/exercism/user.rb

```diff
@@ -148,6 +148,14 @@ def invite_to_track_mentor(language)
     update_acls_for_track_mentor(language)
   end
 
+  def public_user_attributes
+    result = {}
+    result[:username] = self.username
+    result[:avatar_url] = self.avatar_url
+    result[:github_id] = self.github_id
+    result
```

Comment:
This is a take-it-or-leave-it style thing. I would tend to leave off the `self`, since it's implied, and I wouldn't assign this to a temporary variable since the name `result` doesn't communicate anything useful. I would just return the hash directly:

```
{
  username: username,
  avatar_url: avatar_url,
  github_id: github_id,
}
```